File: Go/ExampleTests/common/addresscomponents_string.go
package common

type AddressComponentsString struct {
	Zip string
	StreetName string
	City string
	PreDirection string
	SuffixDirection string
	State string
	SuffixType string
}

func NewAddressComponentsStringFromSlice(v []string) AddressComponentsString {
	s := AddressComponentsString{}
	if len(v) > 0 { s.Zip = v[0] }
	if len(v) > 1 { s.StreetName = v[1] }
	if len(v) > 2 { s.City = v[2] }
	if len(v) > 3 { s.PreDirection = v[3] }
	if len(v) > 4 { s.SuffixDirection = v[4] }
	if len(v) > 5 { s.State = v[5] }
	if len(v) > 6 { s.SuffixType = v[6] }
	return s
}

// NewAddressComponentsStringFromText builds from the text form, e.g. Money as "25 USD".
func NewAddressComponentsStringFromText(text string) AddressComponentsString {
	parts := RequireTokens(text, 7, "AddressComponents")
	return AddressComponentsString{
		Zip: parts[0],
		StreetName: parts[1],
		City: parts[2],
		PreDirection: parts[3],
		SuffixDirection: parts[4],
		State: parts[5],
		SuffixType: parts[6],
	}
}

func (s AddressComponentsString) String() string {
	return Token(s.Zip) + " " + Token(s.StreetName) + " " + Token(s.City) + " " + Token(s.PreDirection) + " " + Token(s.SuffixDirection) + " " + Token(s.State) + " " + Token(s.SuffixType)
}

func (s AddressComponentsString) Equals(o AddressComponentsString) bool {
	return DNCEqual(s.Zip, o.Zip) &&
		DNCEqual(s.StreetName, o.StreetName) &&
		DNCEqual(s.City, o.City) &&
		DNCEqual(s.PreDirection, o.PreDirection) &&
		DNCEqual(s.SuffixDirection, o.SuffixDirection) &&
		DNCEqual(s.State, o.State) &&
		DNCEqual(s.SuffixType, o.SuffixType)
}

func EqualAddressComponentsStringSlices(a, b []AddressComponentsString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/addresscomponents_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct AddressComponentsString {
    pub zip: String,
    pub streetname: String,
    pub city: String,
    pub predirection: String,
    pub suffixdirection: String,
    pub state: String,
    pub suffixtype: String,
}

impl AddressComponentsString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            zip: v.get(0).copied().unwrap_or("").to_string(),
            streetname: v.get(1).copied().unwrap_or("").to_string(),
            city: v.get(2).copied().unwrap_or("").to_string(),
            predirection: v.get(3).copied().unwrap_or("").to_string(),
            suffixdirection: v.get(4).copied().unwrap_or("").to_string(),
            state: v.get(5).copied().unwrap_or("").to_string(),
            suffixtype: v.get(6).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 7, "AddressComponents");
        Self {
            zip: parts[0].clone(),
            streetname: parts[1].clone(),
            city: parts[2].clone(),
            predirection: parts[3].clone(),
            suffixdirection: parts[4].clone(),
            state: parts[5].clone(),
            suffixtype: parts[6].clone(),
        }
    }
}

impl std::fmt::Display for AddressComponentsString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.zip), crate::common::tokens::token(&self.streetname), crate::common::tokens::token(&self.city), crate::common::tokens::token(&self.predirection), crate::common::tokens::token(&self.suffixdirection), crate::common::tokens::token(&self.state), crate::common::tokens::token(&self.suffixtype)].join(" "))
    }
}

impl PartialEq for AddressComponentsString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.zip, &other.zip)
            && crate::common::dnc_equal(&self.streetname, &other.streetname)
            && crate::common::dnc_equal(&self.city, &other.city)
            && crate::common::dnc_equal(&self.predirection, &other.predirection)
            && crate::common::dnc_equal(&self.suffixdirection, &other.suffixdirection)
            && crate::common::dnc_equal(&self.state, &other.state)
            && crate::common::dnc_equal(&self.suffixtype, &other.suffixtype)
    }
}
